workflows/build-base-on-changes.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/python3
+
+import apt_pkg
+import gzip
+import logging
+import os
+import re
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+import urllib.request
+import yaml
+
+from argparse import ArgumentParser
+from datetime import datetime
+from launchpadlib.launchpad import Launchpad
+
+
+SNAP_API = \
+    'https://api.launchpad.net/devel/~ubuntu-core-service/+snap/core{}'
+CORE_SNAP_API = \
+    'https://api.launchpad.net/devel/~snappy-dev/+snap/core{}'
+
+
+_logger = logging.getLogger('ubuntu-image')
+
+# The empty string stands for the core snap, which was released as 16 series
+series_map = {
+    "": "xenial",
+    "18": "bionic",
+    "20": "focal",
+    "22": "jammy",
+    "24": "noble",
+}
+
+
+def update_snap2version(snap2version, package, version):
+    if version.isspace() and package.isspace():
+        return
+    if version.isspace() or package.isspace():
+        print('parse error, one of package ({}) or version ({}) is empty'.
+              format(package, version))
+        sys.exit(1)
+    # Update if new version is more modern only
+    if package in snap2version:
+        if apt_pkg.version_compare(version, snap2version[package]) > 0:
+            snap2version[package] = version
+    else:
+        snap2version[package] = version
+
+
+def package_versions_from_file(pkgs_p, snap2version):
```

Comment:
Could we use `debian.deb822.Packages` from https://salsa.debian.org/python-debian-team/python-debian ?

Reply:
Good point, I've changed this now